scio-snowflake/src/main/scala/com/spotify/scio/snowflake/SnowflakeIO.scala

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2024 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.snowflake
+
+import scala.util.chaining._
+import com.spotify.scio.ScioContext
+import com.spotify.scio.coders.{Coder, CoderMaterializer}
+import com.spotify.scio.io.{EmptyTap, EmptyTapOf, ScioIO, Tap, TapT, TestIO}
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import kantan.csv.{RowCodec, RowDecoder, RowEncoder}
+import org.apache.beam.sdk.io.snowflake.SnowflakeIO.{CsvMapper, UserDataMapper}
+import org.apache.beam.sdk.io.snowflake.data.SnowflakeTableSchema
+import org.apache.beam.sdk.io.snowflake.enums.{CreateDisposition, WriteDisposition}
+import org.apache.beam.sdk.io.{snowflake => beam}
+import org.joda.time.Duration
+
+object SnowflakeIO {
+
+  final def apply[T](opts: SnowflakeConnectionOptions, query: String): SnowflakeIO[T] =
+    new SnowflakeIO[T] with TestIO[T] {
+      final override val tapT = EmptyTapOf[T]
+      override def testId: String = s"SnowflakeIO(${snowflakeIoId(opts, query)})"
+    }
+
+  private[snowflake] def snowflakeIoId(opts: SnowflakeConnectionOptions, target: String): String = {
+    // source params
+    val params = Option(opts.database).map(db => s"db=$db") ++
+      Option(opts.warehouse).map(db => s"warehouse=$db")
```

Comment:
Comparing this to other complex `testId` implementations in our Scio IOs (example [1](https://github.com/spotify/scio/blob/v0.14.9/scio-google-cloud-platform/src/main/scala/com/spotify/scio/pubsub/PubsubIO.scala#L174), [2](https://github.com/spotify/scio/blob/v0.14.9/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala#L254)), I think a format like this would fit better:

```scala
SnowflakeIO(url, target, warehouse?, db?)
```

wdyt @RustedBones

Reply:
Correct! we don't use named params. Will change that